garage/envs/mujoco/mujoco_env.py

```diff
@@ -120,7 +120,7 @@ def reset_mujoco(self, init_state=None):
                 datum = getattr(self.sim.data, datum_name)
                 datum_dim = datum.shape[0]
                 datum = init_state[start:start + datum_dim]
-                setattr(self.sim.data, datum_name, datum)
+                exec('self.sim.data.%s[:] = datum' % datum_name)
```

Comment:
### The problem
This bug is caused by mujoco_py making the references to the underlying numpy arrays in `self.sim.data` read-only. You are free to write into the underlying array (`self.sim.data.ctrl[:] = self.init_ctrl`), but you may not change the array to which the attr points (`self.sim.data.ctrl = self.init_ctrl`)

### How to fix it
I believe your line is equivalent to
```python
getattr(self.sim.data, datum_name).__setitem__(slice(None), datum)
# same as
self.sim.data.datum_name[:] = datum
# also the same as
exec('self.sim.data.%s[:] = datum' % datum_name)
```
which is ugly, but at least it doesn't use `exec()`

### I would prefer if you just unrolled the `for` loop and wrote everything out. 
The original code is some obscure Python meta-programming, and it's difficult to read. There are exactly 4 well-known elements to cover, and nobody would ever be confused about what's going on if it was written out (which is not the case at all for the original version). 

### Comments
* please don't use `exec()` in almost all cases. meta-programming like that is very hard to test and this case is not exceptional-enough to warrant it.

Reply:
@ryanjulian Thank you so much for the comments and I learnt a lot from you! I'll patch the fix right now!